.github/workflows/reusable-release-integration.yml

```diff
@@ -0,0 +1,266 @@
+        with:
+          username: ${{ secrets.dockerhub_username }}
+          password: ${{ secrets.dockerhub_token }}
+      - name: List files
+        run: ls -la
+      - name: Build and load x64 image for security scanning
+        # We need to build a single-arch image again to be able to --load it into the host
+        run: |
+          docker buildx build --load --platform=linux/amd64 \
+            -t $DOCKER_IMAGE_NAME:ci-scan \
+            .
+      - name: Build and push docker prerelease image
+        if: ${{ github.event.release.prerelease }}
+        run: |
+          DOCKER_IMAGE_TAG=${DOCKER_IMAGE_TAG}-pre
+          docker buildx build --push --platform=$DOCKER_PLATFORMS \
+            --build-arg "COMMIT=$COMMIT" \
+            --build-arg "DATE=$DATE" \
+            --build-arg "TAG=$DOCKER_IMAGE_TAG" \
+            -t $DOCKER_IMAGE_NAME:$DOCKER_IMAGE_TAG \
+            .
+      - name: Build and push docker release image
+        if: ${{ ! github.event.release.prerelease }}
+        run: |
+          docker buildx build --push --platform=$DOCKER_PLATFORMS \
+            --build-arg "COMMIT=$COMMIT" \
+            --build-arg "DATE=$DATE" \
+            --build-arg "TAG=$DOCKER_IMAGE_TAG" \
+            -t $DOCKER_IMAGE_NAME:$DOCKER_IMAGE_TAG \
+            -t $DOCKER_IMAGE_NAME:latest \
+            .
+  open-pr:
+    name: Update version and appVersion and open pr
+    needs: [ docker-integration ]
+    runs-on: ubuntu-latest
+    # run only for releases (not prereleases)
+    if: ${{ ! github.event.release.prerelease && inputs.enable_helm_chart_release }}
+    steps:
+      - name: Checkout original repo
+        uses: actions/checkout@v4
+        with:
+          repository: ${{ env.ORIGINAL_REPO_NAME }}
+          ref: main
+
+      - name: Find new appVersion
+        id: find-version
+        run: |
+          echo "NEW_APP_VERSION=${{ needs.docker-integration.outputs.new-version }}" >> $GITHUB_ENV
+          echo "new app version: $NEW_APP_VERSION"
+
+      - name: Find current appVersion
+        id: original_version
+        run: |
+          ORIGINAL_APP_VERSION=$(yq eval ${{ inputs.image_name_key }} ${{ inputs.chart_directory }}/Chart.yaml)
+          echo "original app version: $ORIGINAL_APP_VERSION"
+          echo "ORIGINAL_APP_VERSION=$ORIGINAL_APP_VERSION" >> $GITHUB_ENV
+
+      - name: Find current helm chart version
+        run: |
+          CURRENT_VERSION=$(yq eval '.version' ${{ inputs.chart_directory }}/Chart.yaml)
+          echo "version: $CURRENT_VERSION"
+          echo "CURRENT_VERSION=$CURRENT_VERSION" >> $GITHUB_ENV
+
+      - name: Set up Golang
+        uses: actions/setup-go@v4
+        with:
+          go-version: 1.19.11
+
+      - name: Checkout version-update.go app
+        uses: actions/checkout@v4
+        with:
+          repository: newrelic/k8s-metadata-injection
+          path: tools
+          ref: main
+          sparse-checkout: |
+            version-update.go
```

Comment:
I think we should move this file: https://github.com/newrelic/k8s-metadata-injection/blob/main/src/utils/version-update.go to this k8s-automation-repo since we're moving all reusable workflows to this repo so it would be good to keep things together in one repo.

Reply:
I agree, nice catch. Done